### ISSUE TITLE
recon images is black

### ISSUE THREAD

**Issue by syyxsxx** [lucidrains/magvit2-pytorch#31]
i train magvit2 on imagenet
the recon images is black, but the loss  looks normal
![sampled 18 (1)](https://github.com/lucidrains/magvit2-pytorch/assets/32666364/87a33d78-931f-4099-8e0f-7ab26e47ae67)
![image](https://github.com/lucidrains/magvit2-pytorch/assets/32666364/91d1c194-e299-4269-88c2-75b92b338ca7)

**Comment by nangying1112** [comment]
I met the same problem. Have you solved this?

**Comment by dyflional10** [comment]
> I met the same problem. Have you solved this?
I also encountered the same problem where the loss remains constant, and I discovered that the gradient is NaN.

**Comment by syyxsxx** [comment]
I turned off mixed precision training，solve this problem

**Comment by lucidrains** [comment]
@syyxsxx thank you for finding and sharing the source of the issue. i'll figure out what's the matter with mixed precision..

**Comment by dyflional10** [comment]
@lucidrains Thanks for your great work. I turned off mixed precision training, the loss looks normal, but the image reconstruction is abnormal.
![sampled 273](https://github.com/lucidrains/magvit2-pytorch/assets/151718255/414deb7c-acb8-4cab-b381-93c515da5b7b)
![截屏2024-01-24 11 00 03](https://github.com/lucidrains/magvit2-pytorch/assets/151718255/ff606bd8-0cc7-4dcd-9a6f-bb7d7d529baa)

**Comment by misery0424** [comment]
> I turned off mixed precision training，solve this problem

can you share your config, thanks

**Comment by VoyageWang** [comment]
I also met the same problem the construction image is not so normal. Can you share the details of solving this problem? How to turn off the mixed precision traning

> I turned off mixed precision training，solve this problem

**Comment by kxgong** [comment]
Hi, I have a question. Why use images to train magvit-2? Should it designed for videos?
